server/account/account.go

```diff
@@ -13,8 +17,65 @@ const (
 
 type AccountID [HashSize]byte
 
-func New(pk [pki.PubKeySize]byte) AccountID {
+// NewID generates a unique account id with the provided public key bytes.
+func NewID(pk [pki.PubKeySize]byte) AccountID {
 	// hash the pubkey hash(hash(pubkey))
 	h := HashFunc(pk[:])
 	return HashFunc(h[:])
 }
+
+// Account represents a dex account.
+type Account struct {
+	ID      AccountID
+	PrivKey *secp256k1.PrivateKey
+}
+
+// NewAccount creates a new dex account. The associated private key of the
+// account is generated randomly.
+func NewAccount() (*Account, error) {
+	privKey, err := secp256k1.GeneratePrivateKey()
+	if err != nil {
+		return nil, err
+	}
+
+	pubKey := privKey.PubKey()
+
+	var pkb [pki.PubKeySize]byte
+	copy(pkb[:], pubKey.SerializeCompressed()[:pki.PubKeySize])
+	return &Account{
+		ID:      NewID(pkb),
+		PrivKey: privKey,
+	}, nil
+}
+
+// NewAccountFromBytes create a new dex account from the provided private
+// key bytes.
+func NewAccountFromBytes(privK []byte) (*Account, error) {
+	if privK == nil {
+		return nil, fmt.Errorf("invalid input: private key bytes is nil")
+	}
+
+	privKey, pubKey := secp256k1.PrivKeyFromBytes(privK)
+
+	var pkb [pki.PubKeySize]byte
+	copy(pkb[:], pubKey.SerializeCompressed()[:pki.PubKeySize])
+	return &Account{
+		ID:      NewID(pkb),
+		PrivKey: privKey,
+	}, nil
+}
+
+// Sign generates a signature for the provided message.
+func (acc *Account) Sign(msg []byte) ([]byte, error) {
+	if acc.PrivKey == nil {
+		return nil, fmt.Errorf("account not initialized")
+	}
+
+	hash := chainhash.HashB(msg)
```

Comment:
If we use chainhash, we'll need to update the spec to reflect a 14 round Blake-256.  Is there a strong case for 14 round instead of just a single blake256 hash?